oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java

```diff
@@ -50,19 +50,27 @@ public String getState() {
     }
 
     public static AuthorizationResponse fromString(final String s) throws AuthorizationException {
+        return fromString(s, null);
+    }
+
+    public static AuthorizationResponse fromString(final String s, final String potentialDescription) throws AuthorizationException {
         String code = null;
         String state = null;
         String error = "unknown_error";
-        String errorDescription = null;
+        String errorDescription = potentialDescription;
         String errorUriString = null;
-        if (s != null) {
+        if (s != null && s.length() > 0) {
             final String[] pairs = PAIR_SEPARATOR.split(s.trim());
 
             for (final String pair : pairs) {
                 final String[] nameAndValue = NAME_VALUE_SEPARATOR.split(pair, 2);
                 try {
                     if (nameAndValue.length != 2) {
-                        throw new AuthorizationException("parsing_error", "Failed to parse server response", null, null);
+                        final StringBuilder sb = new StringBuilder("Failed to parse server response");
+                        if (errorDescription != null) {
+                            sb.append(".\nDetails: ").append(errorDescription);

```

Comment:
> however, it seems prudent to include the contents of  s  that could not be parsed

Agree it's helpful, but I am afraid we may include too much -- if it contains sensitive information (like the code, or part of it)?    

> As for  stderr , I expect errors there, by definition. We could change the wording to not imply so strongly that the contents of  stderr  directly explain those of  stdout . What do you think?

That's perfect.  We could say "we also got the following stderr, it may or may not be related to the parsing error you are experiencing".. something along this line would be perfect.